### PR TITLE
fix: 10967 User able to add Ledger account with existing account name

### LINKED ISSUE #10967: [Bug] User able to add Ledger account with existing account name
Bitrise Build: https://app.bitrise.io/app/be69d4368ee7e86d/build/d217a4db-d343-43ed-887d-c7dc103cb9f2/artifact/308fa13fdcadc0dc/p/7db511ad6b94e0a1e04bc9bea869a583

Device: iOS 17.6.1

1. Install MM
2. Recover from SRP
3. Rename 'Account 1' to 'Ledger 1'
4. Connect Ledger device
5. Add 1 Ledger account
6. Both accounts are displayed on Account list

Expected: Error displays 'Account name already exists' and user unable to add existing account name
Actual:

### PR DESCRIPTION
**Pull request by dawnseeker8**
## **Description**

This PR fix #10967 User able to add Ledger account with existing account name.
Now the ledger Select accounts screen will display `Account Ledger 1 already exists`

## **Related issues**

Fixes: #10967 
#10188 

## **Manual testing steps**

1. Install MM
2. Recover from SRP
3. Rename 'Account 1' to 'Ledger 1'
4. Connect Ledger device
5. Add 1 Ledger account
6. `Account Ledger 1 already exists` error should be displayed

### **Before**
image

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by vivek-consensys** [comment]
Bitrise [build](https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/21fa0d1a-d26c-4923-8529-a411668521cb?tab=workflows) 

Tested on iPhone 13 / iOS 18.1

[Fix](https://github.com/MetaMask/metamask-mobile/issues/10967) 


https://github.com/user-attachments/assets/690205ad-ddb2-4f14-8afb-86815a20fc66

**Comment by vivek-consensys** [comment]
Tested using the latest ledger library [upgrade](https://github.com/MetaMask/metamask-mobile/pull/11836) on bitrise [build](https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/957add45-66f2-4d8e-b79a-df93bba4f358?tab=artifacts) and working as expected.

File: app/core/Ledger/Ledger.ts [hunk 1]
```diff
@@ -208,14 +209,45 @@ export const ledgerSignTypedMessage = async (
...
 export const unlockLedgerWalletAccount = async (index: number) => {
-  await withLedgerKeyring(async (keyring: LedgerKeyring) => {
+  const accountsController = Engine.context.AccountsController;
+  const { unlockAccount, name}  = await withLedgerKeyring(async (keyring: LedgerKeyring) => {
+    const existingAccounts = await keyring.getAccounts();
+    const keyringName = keyringTypeToName(ExtendedKeyringTypes.ledger);
+    const accountName = `${keyringName} ${existingAccounts.length + 1}`;
+
+    if(await checkAccountNameExists(accountName)) {
+      throw new Error(`Account ${accountName} already exists`);
```
Comment (joaoloureirop) [inline app/core/Ledger/Ledger.ts:237]: Should add error message to translation file `locales/languages/en.json`

**Review by joaoloureirop — CHANGES_REQUESTED** [review]
code lgtm.

Can we have wallet ux approval on blocking users from importing Ledger wallets until they manually rename the existing non-ledger wallet?

I'm concerned this friction might upset users.
We could offer a wallet rename at this point.

File: app/core/Ledger/Ledger.ts (hunk 1 above)
Comment (dawnseeker8) [reply app/core/Ledger/Ledger.ts]: have make the change to add translation key into en.json file.

**Review by joaoloureirop — APPROVED** [review]

**Pull request merged**